### PR TITLE
feat(build): add aarch64 release target

### PR DESCRIPTION
**Pull request by tychoish**
The `justfile` already has support. 

I'd definitely like someone with a bit more rust toolchain familiarty
to gut check my assumption that this will _just work_ on such an old
version of ubuntu.

### CODE REVIEW

**Review by scsmithr — COMMENTED** [review]
If it compiles, I'm sure it'll work. Old version of ubuntu is because of glibc: https://github.com/GlareDB/glaredb/pull/1739

File: .github/workflows/release.yaml
Comment (scsmithr) [inline .github/workflows/release.yaml:21]: We might need to create a machine type for this. The existing 8 core ones are custom large runners.

Comment (tychoish) [reply .github/workflows/release.yaml:21]: how do we do that?

**Review by zhangping99 — APPROVED** [review]

**Comment by tychoish** [comment]
Just to follow up, I think @scsmithr needs to add the updated builder to the GlareDB org to unblock this.

File: .github/workflows/release.yaml
Comment (scsmithr) [reply .github/workflows/release.yaml:21]: Screenshot 2024-01-05 at 1 03 19 PM

runner added

**Comment by tychoish** [comment]
Seems to be hitting a link error:

https://github.com/GlareDB/glaredb/actions/runs/7438394256/job/20237097364#step:6:1315

**Comment by tychoish** [comment]
I've poked at this and my current assessment is that: there's some binary blob (or something) and/or it's falling back to using the wrong linker.

I can reproduce this locally but I haven't convinced cargo/rustc to do the right thing.

**Comment by universalmind303** [comment]
marking this as draft as I don't think this is actively waiting on review

**Comment by tychoish** [comment]
upstream dependencies are blocking this more long term, and our current plan is to address this by replacing them or making them optional. Going to close this and revisit later.

**Pull request closed without merging**
